### ISSUE TITLE
Distinct mac addresses in Utilization session query

### ISSUE THREAD

**Issue by gmazlami** [probr/probr-analysis#54]

**Comment by gmazlami** [comment]
Is this really an issue? By the way the session computation in the worker is defined, there are no overlapping sessions for the same mac_address, otherwise those would be merged by the session job. Hence there is no need for "distinct" mac adress querying.

**Comment by sebastian-stephan** [comment]
Yes this is an issue. There might be non-overlapping sessions from the same mac in the same bucket. I.e. from 13:36-13:37 and from 13:38-13:39. This would currently result in the count of two sessions.

This is the smallest problem with the chart at the moment. There are several bugs. I wrote a pretty complete test for the /api/sessions/reduce endpoint to highlight them and pushed them to dev.

Run tests with:

``` sh
npm test
```

Output currently:

```
  /api/session/reduce endpoint
    ✓ should be a function
    ✓ should return non empty array
    ✓ should return {_id: Date, value: Number} objects
    ✓ should return dates that are rounded down to 5 minutes
    1) should report empty buckets in the middle
    2) should report empty buckets to the left and right of the requested range
    3) should report edge buckets correctly
    single sessions
      ✓ should correctly report a session that completely covers a bucket
      4) should correctly report a session that starts inside bucket
      ✓ should correctly report a session that ends inside bucket
    overlapping sessions
      ✓ should correctly report two sessions that both overlap on an entire bucket
      5) should correctly report two sessions where one starts in the bucket and the other covers it
      ✓ should correctly report two sessions where one ends in the bucket and the other covers it
    multiple sessions in one bucket
      6) should only generate one value if two sessions in one bucket belong to the same mac
      7) should generate two values if two sessions in one bucket belong to different macs
```

Check out [session.controller.spec.js](https://github.com/probr/probr-analysis/blob/dev/server/api/session/session.controller.spec.js)
